### PR TITLE
Fix an issue in db connect function

### LINKED ISSUE #1144: [Multi-ESX] DB connection issue while invoking "vmdkops_admin volume ls" observed only on one ESX
Setup: VM1 on ESX1 and VM2 on ESX2 > both docker hosts are in the same tenant

Steps to reproduce
```
1. VM1: create volume vol4
2. VM1 & VM2: docker volume ls > make sure vol1 is listed out
3. VM1: attach container on vol4 and keep it running
4. VM2: attach the container on vol4 (observe some wait time here)
5. Expects an error as vol1 is already mounted and in use
```

While invoking ```vmdkops_admin.py volume ls``` during step#4, running into following issue on ESX1 where as ESX2 is not complaining and showing the output correctly.

**Note**: I haven't found any interesting thing in vmdk_ops.log so not uploading. (logs will be uploaded upon ask)

```
[root@sc-rdops-vm08-dhcp-230-54:~] /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py status
=== Service: 
Version: 0.12.9e2d6ad-0.0.1
Status: Running
Pid: 4257207
Port: 1019
LogConfigFile: /etc/vmware/vmdkops/log_config.json
LogFile: /var/log/vmware/vmdk_ops.log
LogLevel: INFO
=== Authorization Config DB: 
DB_SharedLocation: /vmfs/volumes/vsanDatastore/dockvols/vmdkops_config.db
DB_LocalPath: /etc/vmware/vmdkops/auth-db
DB_Mode: MultiNode (local symlink pointing to shared DB)
```

on ESX1:
```
[root@sc-rdops-vm08-dhcp-230-54:~] /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py volume ls
Traceback (most recent call last):
  File "/usr/lib/vmware/vmdkops/bin/vmdkops_admin.py", line 1356, in <module>
    main()
  File "/usr/lib/vmware/vmdkops/bin/vmdkops_admin.py", line 58, in main
    args.func(args)
  File "/usr/lib/vmware/vmdkops/bin/vmdkops_admin.py", line 593, in ls
    rows = generate_ls_rows(tenant_reg)
  File "/usr/lib/vmware/vmdkops/bin/vmdkops_admin.py", line 622, in generate_ls_rows
    for v in vmdk_utils.get_volumes(tenant_reg):
  File "/usr/lib/vmware/vmdkops/Python/vmdk_utils.py", line 147, in get_volumes
    error_info, tenant_name = auth_api.get_tenant_name(sub_dir_name)
  File "/usr/lib/vmware/vmdkops/Python/auth_api.py", line 68, in get_tenant_name
    error_info, auth_mgr = get_auth_mgr_object()
  File "/usr/lib/vmware/vmdkops/Python/auth_api.py", line 38, in get_auth_mgr_object
    err_msg, auth_mgr = auth.get_auth_mgr()
  File "/usr/lib/vmware/vmdkops/Python/auth.py", line 46, in get_auth_mgr
    thread_local._auth_mgr.connect()
  File "/usr/lib/vmware/vmdkops/Python/auth_data.py", line 572, in connect
    self.__mode = self.__discover_mode_and_connect()
  File "/usr/lib/vmware/vmdkops/Python/auth_data.py", line 610, in __discover_mode_and_connect
    self.__connect()
  File "/usr/lib/vmware/vmdkops/Python/auth_data.py", line 559, in __connect
    self.conn = sqlite3.connect(self.db_path)
sqlite3.OperationalError: unable to open database file
```
on ESX2
```
[root@sc-rdops-vm08-dhcp-239-65:~] /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py volume ls
Volume    Datastore    VM-Group  Capacity  Used  Filesystem  Policy  Disk Format  Attached-to  Access      Attach-as               Created By    Created Date              
--------  -----------  --------  --------  ----  ----------  ------  -----------  -----------  ----------  ----------------------  ------------  ------------------------  
vol1      shared_vmfs  _DEFAULT  100MB     15MB  ext4        N/A     thin         VM1          read-only   independent_persistent  VM1           Wed Mar 29 23:25:37 2017  
vol2      shared_vmfs  _DEFAULT  100MB     13MB  ext4        N/A     thin         detached     read-write  independent_persistent  VM1           Wed Mar 29 23:26:23 2017  
vol3      shared_vmfs  _DEFAULT  100MB     13MB  ext4        N/A     thin         detached     read-write  independent_persistent  VM1           Wed Mar 29 23:27:18 2017  
vol4      shared_vmfs  _DEFAULT  100MB     15MB  ext4        N/A     thin         detached     read-write  independent_persistent  VM1           Mon Apr  3 21:48:33 2017  
vol5      shared_vmfs  _DEFAULT  100MB     13MB  ext4        N/A     thin         detached     read-write  independent_persistent  photon-VM0.1  Mon Apr  3 22:13:37 2017  
```

/CC @msterin @lipingxue

### PR DESCRIPTION
**Pull request by shaominchen**
sqlite3.connect() function will throw an exception instead of returning none if running into any errors. So we should try/catch the error instead of checking for none result.

Fix #1144 

Test result:
1. Set up MultiNode on 2 ESXs
2. Open the DB on ESX1
3. Run "vmdkops_admin vmgroup ls" on ESX2

Error from step 3: 
[root@sc-rdops-vm09-dhcp-28-64:/var/log/vmware] vmdkops_admin vmgroup ls
Internal Error(DB connection error at /etc/vmware/vmdkops/auth-db)

Error from vmdk_ops.log:
05/03/17 00:15:41 91741 [MainThread] [ERROR  ] Failed to connect to DB (/etc/vmware/vmdkops/auth-db): unable to open database file

### CODE REVIEW

**Comment by shaominchen** [comment]
Note: I'm still waiting for my new testbed to be ready to verify the fix.

**Comment by msterin** [comment]
LGTM pending usual stuff:
 (1) test: need a cut-n-paste example of triggering the exception (2) need CI pass when  CI is back

**Comment by shuklanirdesh82** [comment]
@shaominchen Please rebase your branch with master to get .drone.sec changes.

File: esx_service/utils/auth_data.py [hunk 1]
```diff
@@ -630,14 +630,18 @@ def __handle_upgrade(self):
...
         if self.conn:
-            logging.info("AuthorizationDataManagerReconnecting to %s on request", self.db_path)
+            logging.info("AuthorizationDataManager: Reconnecting to %s on request", self.db_path)
             self.__close()
-        self.conn = sqlite3.connect(self.db_path)
-        if not self.conn:
+
+        try:
+            self.conn = sqlite3.connect(self.db_path)
+        except sqlite3.Error as e:
```
Comment (pshahzeb) [inline esx_service/utils/auth_data.py]: nit: we can log the content of object e as well.

Comment (shaominchen) [inline esx_service/utils/auth_data.py]: Sure, will log the error.

**Comment by pshahzeb** [comment]
Not this PR; but I feel all the errors reported to user/admin shouldn't be marked as "internal error". They should be clear as to what error or type of error is.

**Pull request merged**